Implement home-manager support (#168)
Mic92/sops-nix: README.md

}
```

> ## Use with home manager

Reviewer: Reminder to myself to expose this in the sops flake as well.

sops-nix also provides a home-manager module.
This module provides a subset of features provided by the system-wide sops-nix since features like the creation of the ramfs and changing the owner of the secrets are not available for non-root users.

Instead of running as an activation script, sops-nix runs as a systemd user service called `sops-nix.service`.
And instead of decrypting to `/run/secrets`, the secrets are decrypted to `$XDG_RUNTIME_DIR/secrets`.

> **Since the secrets are decryted there, it's highly recommended to use a tmpfs for `$XDG_RUNTIME_DIR` if your distribution does not do that.**

Reviewer: ```suggestion
**Since the secrets are decrypted there, it's highly recommended to use a tmpfs for `$XDG_RUNTIME_DIR` to avoid storing secrets in plain text on persistent storage. Linux distributions using systemd-logind do that out-of-the-box.**
```

Depending on whether you use home-manager system-wide or using a home.nix, you have to import it in a different way.
This example show the `channel` approach from the usage example above for simplicity, but all other methods work as well.

```nix
{
  # NixOS system-wide home-manager configuration
  home-manager.sharedModules = [
    <sops-nix/modules/home-manager/sops.nix>
  ];
}
```

```nix
{
  # Configuration via home.nix
  imports = [
    <sops-nix/modules/home-manager/sops.nix>
  ];
}
```

The actual sops configuration is in the `sops` namespace in your home.nix (or in the `home-manager.users.<name>` namespace when using home-manager system-wide):
```nix
{
  sops = {

>     age.keyFile = "/home/user/.age-key.txt" ]; # must have no password!

Reviewer 2: ```suggestion
    age.keyFile = "/home/user/.age-key.txt"; # must have no password!
```

    # It's alos possible to use a ssh key, but only when it has no password:
    #age.sshKeyPaths = [ "/home/user/path-to-ssh-key" ];

>     sops.secrets.test = {
>       sopsFile = ./secrets.yml.enc;

Reviewer 2: ```suggestion
    defaultSopsFile = ./secrets.yml.enc;
    sops.secrets.test = {
      # sopsFile = ./secrets.yml.enc; # optionally define per-secret files
```

>     #age.sshKeyPaths = [ "/home/user/path-to-ssh-key" ];
>     sops.secrets.test = {

Reviewer 2: ```suggestion
    #age.sshKeyPaths = [ "/home/user/path-to-ssh-key" ];
    defaultSopsFile = ./secrets.yaml;
    sops.secrets.test = {
```

Reviewer 2: this is required, but got lost somehow

}
```

## Use with home manager

sops-nix also provides a home-manager module.
This module provides a subset of features provided by the system-wide sops-nix since features like the creation of the ramfs and changing the owner of the secrets are not available for non-root users.

Instead of running as an activation script, sops-nix runs as a systemd user service called `sops-nix.service`.

> And instead of decrypting to `/run/secrets`, the secrets are decrypted to `$XDG_RUNTIME_DIR/secrets`.

Reviewer: `$XDG_RUNTIME_DIR` can be not set on non-nixos. We should recommend to have it point it to a tmpfs so that secrets are not persisted stored in plain text or otherwise live with the consequences...

> And instead of decrypting to `/run/secrets`, the secrets are decrypted to `$XDG_RUNTIME_DIR/secrets`.

Reviewer: `$XDG_RUNTIME_DIR` can be not set on non-nixos. We should recommend to have it point it to a tmpfs so that secrets are not persisted stored in plain text or otherwise live with the consequences...

Usage example:
```nix
{
  # NixOS home-manager configuration
  home-manager.sharedModules = [

>     /path/to/sops-nix/modules/home-manager/sops.nix

Reviewer: Maybe fetchTarball could work here? I would also expose the module via flake.nix.

>     /path/to/sops-nix/modules/home-manager/sops.nix

Reviewer: Maybe fetchTarball could work here? I would also expose the module via flake.nix.

  ];

  # Configuration via home.nix
  imports = [
    /path/to/sops-nix/modules/home-manager/sops.nix

>   ];

Reviewer: It's confusing that home-manager and nixos configuration are in one code block. I would also keep the `home-manager` configuration below separate from importing sops-nix.

>   ];

Reviewer: It's confusing that home-manager and nixos configuration are in one code block. I would also keep the `home-manager` configuration below separate from importing sops-nix.

  # Configuration of secrets
  sops = {

>     age.sshKeyPaths = [ "/home/user/path-to-ssh-key" ]; # must have no password!

Reviewer: I am inclined to make normal age keys the default here as a best practice. Most of the time you want to have ssh keys encrypted because of their actual use.

>     age.sshKeyPaths = [ "/home/user/path-to-ssh-key" ]; # must have no password!

Reviewer: I am inclined to make normal age keys the default here as a best practice. Most of the time you want to have ssh keys encrypted because of their actual use.

    sops.secrets.test = {
      sopsFile = ./secrets.yml.enc;

>       path = "%r/test.txt"; # %r gets replaced with your $XDG_RUNTIME_DIR

Reviewer: In that case we should also replace `%%` with `%` without expanding it for escaping purpose.

```suggestion
      path = "%r/test.txt"; # %r gets replaced with your $XDG_RUNTIME_DIR
```

>       path = "%r/test.txt"; # %r gets replaced with your $XDG_RUNTIME_DIR

Reviewer: In that case we should also replace `%%` with `%` without expanding it for escaping purpose.

```suggestion
      path = "%r/test.txt"; # %r gets replaced with your $XDG_RUNTIME_DIR
```
